### PR TITLE
If the user tries to `Pkg.add(; path = ...)` a path that is not a Git repo, suggest `Pkg.develop` instead

### PR DESCRIPTION
**Pull request by DilumAluthge**

### CODE REVIEW

File: src/Types.jl [hunk 1]
```diff
@@ -624,7 +624,10 @@ function handle_repo_add!(ctx::Context, pkg::PackageSpec)
     if !isurl(pkg.repo.source)
         if isdir(pkg.repo.source)
             if !isdir(joinpath(pkg.repo.source, ".git"))
-                pkgerror("Did not find a git repository at `$(pkg.repo.source)`")
+                has_project_file = isfile(joinpath(pkg.repo.source, "Project.toml")) || isfile(joinpath(pkg.repo.source, "JuliaProject.toml"))
+                pkg_develop_suggestion = has_project_file ? ". You may consider using `Pkg.develop` instead" : ""
+                msg = "Did not find a git repository at `$(pkg.repo.source)`$(pkg_develop_suggestion)"
+                pkgerror(msg)
```
Comment (fredrikekre) [inline src/Types.jl:630]:
```suggestion
                msg = "Did not find a git repository at `$(pkg.repo.source)`"
                if !(isfile(joinpath(pkg.repo.source, "Project.toml")) || isfile(joinpath(pkg.repo.source, "JuliaProject.toml")))
                    msg *= ", perhaps you meant `Pkg.develop`?"
                end
                pkgerror(msg)
```

**Review by fredrikekre — COMMENTED** [review]
Could use a test

**Comment by DilumAluthge** [comment]
> Could use a test

Done!

**Comment by DilumAluthge** [comment]
@fredrikekre This is the current behavior in the PR:
1. If you have a Project.toml file, we'll print the `Pkg.develop` suggestion.
2. If you don't have a project file, we won't print the suggestion.

Does that look correct?

**Pull request merged**